### ISSUE TITLE
"No valid characters in data" when uploading data from characters of a specific realm

### ISSUE THREAD

**Issue by danieledelgiudice** [madcowfred/wowthing-issues#45]
When uploading the _WoWthing_Collector.lua_ file to the website I get an error message saying:

> No valid characters in data

 Uploading with the WoWthing Sync app gave a different error message:

> Upload failed: Invalid Lua data file.

After doing some tests I believe it's because my characters are in the realm _Pozzo dell'Eternità_ (EU).
If I let the companion addon collect data from characters on other realms the upload goes through, but my characters on _Pozzo dell'Eternità_ are ignored, and their data is not added on the website.

In the past I've had issues with several websites using the WoW API when loading data from characters from that realm, so I think that might be the issue. Just to let you know, the "english name" of the realm is _Well Of Eternity_.

I'm [attaching the file](https://github.com/madcowfred/wowthing-issues/files/1838719/WoWthing_Collector.lua.txt) containing data from a single character on the that realm: the file was generated from the companion addon (loading it as a .txt only for GitHub purposes). 

Thanks for your time and your work, let me know if you need anything else.

**Comment by madcowfred** [comment]
Apologies for the slow response, I was away from home with no dev environment.

Looks like the database and the upload handler were disagreeing about how to represent the "à" character (and any other non-ASCII characters). I've pushed a tentative fix for this, can you try uploading now?

**Comment by danieledelgiudice** [comment]
At the moment I don't have an active subscription, so I'm sorry if I can't be thorough.
In my tests I used a _WoWthing_Collector.lua_ file from the last time I was online, around 3 weeks ago. 

I've first tried to upload the data using the website. The upload went through, but only some of the _Pozzo Dell'Eternità_ characters data was loaded correctly.

After that I tried uploading using the _WoWthing Sync_ app: I manually changed the file with an irrelevant change, the app noticed it and uploaded the file. Everything went through and all of my characters data was updated (I'm not sure if that's because the server takes some time to propagate the changes).

I hope that helped you. Thanks again for your work!
